### DISCUSSION TITLE
Idea Statica API to perform checks with existing FEA model and results

### DISCUSSION THREAD

**Discussion by Knuttatutta** [idea-statica/ideastatica-public#728] [Help Q&A] [answered]
Hi,

I was trying to find some information about this, but I couldn't find an example that matched our workflow, so I figured I'd come here and ask.

We currently have a workflow where we perform an FEA analysis for the same model in SCIA, ABAQUS and DIANA. Then in Python we extract and process those results from all the different software.

The next step now is to perform checks on all the elements, and for the steel member checks we would like to see if we can use the Idea Statica Python API. 

Would it be possible in the Python API to create the steel members and perform code checks using the results for the internal forces we already have? With the SCIA API we can create the members, but we cannot add existing resulting internal forces.

[This example](https://github.com/idea-statica/ideastatica-public/wiki/Example-Creating-IOM-with-Python-Steel-Frame) might actually cover most of what we need, but the example skips the results. I think I'm also seeing that it's adding load groups, cases and combinations, but I don't see any application of actual loads, so I suppose maybe this is needed anyway to assign the results to the members and to then perform the checks.

So actually it does appear that all of this is possible, but is there maybe a complete example somewhere for this workflow?

Thank you!

**Comment by nathanlukeIDEA** [comment]
Hi @Knuttatutta,

Can you please confirm which IDEA StatiCa Application you want to use?  IDEA StatiCa Member?

Cheers
Nathan

**Comment by Knuttatutta** [comment]
Thanks for the response, @nathanlukeIDEA 

It would be IDEA StatiCa Member. I don't have much experience with it myself as I'm just developer.

Best,
Knut

**Comment by nathanlukeIDEA** [comment, marked as answer]
Hi @Knuttatutta, we do not have capability to create member projects through an API atm. We only have an API for our Connection Application.

**Comment by Knuttatutta** [comment]
Hi @nathanlukeIDEA ,

Really? This is a bit confusing to me considering the example that I linked to and also that there's the IOM API and the RCS API.
I am completely unfamiliar with IdeaStatiCA, but it looks like there are more APIs than just for the Connection Application?

Do you maybe mean that the checks I want to perform are only supported for connections and not steel members?

**Reply by nathanlukeIDEA** [reply]
Hi @Knuttatutta, yes that is correct. IOM is a data format that supports most of our applications, but we cannot import and run calculations/retrieve results through an API at this point for members.

**Reply by Knuttatutta** [reply]
Okay I understand, thank you
